webui/react/src/hooks/useModal/useModalChangePassword.tsx

```diff
@@ -0,0 +1,122 @@
+import { Button, Form, Input } from 'antd';
+import React from 'react';
+
+import { useStore } from 'contexts/Store';
+import { login, setUserPassword } from 'services/api';
+
+import useModal, { ModalHooks } from './useModal';
+import css from './useModalChangePassword.module.scss';
+
+const useModalChangePassword = (): ModalHooks => {
+  const { modalClose, modalOpen: openOrUpdate, modalRef } = useModal();
+  const { auth } = useStore();
+  const username = auth.user?.username || 'Anonymous';
+  const [ form ] = Form.useForm();
+
+  const getModalContent = () => {
+    const submitForm = async (): Promise<void> => {
+      await setUserPassword({
+        password: form.getFieldValue('newPassword'),
+        username,
+      });
+      modalClose();
+    };
+
+    return (
+      <div className={css.base}>
+        <Form form={form} layout="vertical" onFinish={submitForm}>
+          <Form.Item
+            label="Old Password"
+            name="oldPassword"
+            required
+            rules={[
+              {
+                message: 'Incorrect password',
+                validator: async (rule, value) => {
+                  return await login({
+                    password: value || '',
+                    username: username || '',
```

Comment:
nit: above you're setting `username = auth.user?.username || 'Anonymous';`, so why do you need the `|| ''` here?